sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -87,4 +89,35 @@ object StringUtils {
     }
     funcNames.toSeq
   }
+
+  /**
+   * Concatenation of sequence of strings to final string with cheap append method
```

Comment:
I am wondering if `StringConcatenation` is a better name for this class, as this class is technically a rope (that is a binary tree).

Reply:
... and is wrong :P...